src/components/tutorialEditor/stepContent.tsx

```diff
@@ -0,0 +1,237 @@
+  width: 40px;
+  height: 40px;
+  &:hover {
+    cursor: pointer;
+  }
+`;
+
+const TextInput: AnyStyledComponent = styled.textarea`
+  width: 100%;
+  min-height: 82px;
+  background-color: #46464e;
+  outline: none;
+  border: none;
+  border-radius: 1px;
+  padding: 5px;
+  color: #f5f5f5;
+  resize: none;
+`;
+
+const DeleteBlock: AnyStyledComponent = styled.div`
+  display: ${(props: DeleteProps): string => props.visible};
+  border-left: solid 1px #727272;
+  width: 7%;
+  text-align: center;
+  justify-content: center;
+  &:hover {
+    cursor: pointer;
+    background: rgba(255, 255, 255, 0.1);
+  }
+`;
+
+const TrashIcon: AnyStyledComponent = styled(Icon)`
+  align-self: center;
+  width: 22px;
+  height: 22px;
+`;
+
+interface ContentBlock {
+  type: string;
+  value: string;
+}
+
+interface StepContentProps {
+  value: ContentBlock;
+  index: number;
+  content: ContentBlock[];
+  setContent: Function;
+}
+
+const StepContent: React.FC<StepContentProps> = (props): JSX.Element => {
+  const { value, index, content, setContent } = props;
+
+  const [hovering, setHovering] = useState(-1);
+
+  const codeInput = {
+    position: "relative",
+    marginTop: "0px",
+    height: "82px",
+    width: "100%",
+    backgroundColor: background,
+    fontFamily: "Source Code Pro",
+    display: content[index].type === "code" ? "block" : "none"
+  };
+
+  const changeType = (type: string): void => {
+    const contentCopy = [...content];
+    contentCopy[index].type = type;
+    setContent(contentCopy);
+  };
+
+  const changeOrder = (direction: "up" | "down"): void => {
+    const contentCopy = [...content];
+    if (direction === "up" && index - 1 >= 0) {
+      const temp = contentCopy[index - 1];
+      contentCopy[index - 1] = contentCopy[index];
+      contentCopy[index] = temp;
+    } else if (direction === "down" && index + 1 < content.length) {
+      const temp = contentCopy[index + 1];
+      contentCopy[index + 1] = contentCopy[index];
+      contentCopy[index] = temp;
+    }
+    setContent(contentCopy);
+  };
+
+  const deleteBlock = (): void => {
+    const contentCopy = [...content];
+    contentCopy.splice(index, 1);
+    setContent(contentCopy);
+  };
+
+  const handleTextChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ): void => {
+    const contentCopy = [...content];
+    contentCopy[index].value = event.target.value;
+    setContent(contentCopy);
+  };
+
+  const handleCodeChange = (code: string): void => {
+    const contentCopy = [...content];
+    contentCopy[index].value = code;
+    setContent(contentCopy);
+  };
+
+  return (
+    <ContentBlock
+      key={index.toString()}
+      onMouseEnter={(): void => {
+        setHovering(index);
+      }}
+      onMouseLeave={(): void => {
+        setHovering(-1);
+      }}
+    >
+      <ContentBlockInnerWrapper>
+        <Button
+          backgroundColor={
+            value.type === "text" ? "rgba(256, 256, 256, 0.2)" : "none"
```

Comment:
No issues here but one decent advice that I heard from a dev (Ben Awad) is that whenever you find yourself writing code 3 times, it is then finally ready to be it's own function/class/component. Just thought I'd share.